### DISCUSSION TITLE
DNN API help page/documentation

### DISCUSSION THREAD

**Discussion by andrei-curta** [dnnsoftware/Dnn.Platform#4357] [Q&A]
Hello,

I have created a custom API module based on DNN and I am wondering if there is a way of creating a help page for it where the endpoints would be visible and documented. 
I have tried using swagger and adapting the ASP .Net WEB API  help page, but with no luck.

Thank you in advance!

**Comment by valadas** [comment]
Dnn has URL rewriting which I believe will not work out of the box with swagger basic setup instructions. I have not tried this yet and it's on my whish list. But I think changing the SEO setting for ignore urls may do the trick.

This is just a guess, there might be more to it than just that...

**Reply by andrei-curta** [reply]
Thank you for your answer!
I have tried to follow your suggestion, but I did not manage to make it work. I have changed the regular expression under "Do Not Use Friendly URLs Regular Expression" to /Rss\.aspx|/api/MyAPI/swagger. I also used the tool provided within the site settings to check if the url would get rewritten and it showed it wouldn't. I am assuming that swagger should be at http://dnndev.me/API/MyAPI/swagger

The routes in the API are mapped as follows, if it helps:
`public class RouteMapper : IServiceRouteMapper
    {
        public void RegisterRoutes(IMapRoute mapRouteManager)
        {
            mapRouteManager.MapHttpRoute("MyAPI", "default", "{controller}/{action}", new[] { "MyAPI.Controllers" });
        }
    }`

Was this the correct approach?

**Comment by valadas** [comment]
yeah, I did not play with that nearly enough to have a good answer to that, but I am following this thread because I am also interested in this.
